### DISCUSSION TITLE
On change field validation after form is submitted

### DISCUSSION THREAD

**Discussion by bulicmatko** [mantinedev/mantine#1562] [Feedback]
Hello there!

I'm checking Mantine as a possible UI framework for our next project. It looks awesome.
Our UI will have lots of forms and while going through the docs I noticed a bit strange behavior with the form validation. 

- Go to [https://mantine.dev/form/schema/#yup](https://mantine.dev/form/schema/#yup)
- Type `a` to email input.
- Type `a` to name input.
- Submit the form. -> Validation errors will appear on `email` and `name` fields.
- Type a valid email and a name longer than two chars.
- Errors are still there, even though the inputs are now valid.
- Submit the form.
- Errors can still appear if you type i.e. too long name.

I feel that the user experience would be improved if the fields would be validated on change after the form is submitted for the first time.
Then, user would get faster feedback on what's wrong with the input.

Formik and similar form libraries already work in that (or similar) way.

What do you guys think?

PS: Thanks the author and all the maintainers for the great work and awesome library. :)

**Comment by rtivital** [comment]
Thanks for reporting, it is a bug with schema validation, it will be fixed in one of the next patches

**Reply by bulicmatko** [reply]
Ok. Thanks for the fast feedback and for awesome work!

**Reply by rtivital** [reply]
Fixed in `4.2.8`
